ml_peg/calcs/supramolecular/plf547/calc_plf547.py

```diff
@@ -0,0 +1,352 @@
+
+MODELS = load_models(current_models)
+
+KCAL_TO_EV = units.kcal / units.mol
+EV_TO_KCAL = 1 / KCAL_TO_EV
+
+DATA_PATH = Path(__file__).parent / "data"
+
+OUT_PATH = Path(__file__).parent / "outputs"
+DATA_PATH = (
+    download_s3_data(
+        filename="PLF547.zip", key="inputs/supramolecular/PLF547/PLF547.zip"
+    )
+    # / "PLF547"
+)
+print(DATA_PATH)
+
+
+class PLF547Benchmark(zntrack.Node):
+    """Benchmarking PLF547 subset ionic hydrogen bonds benchmark dataset."""
+
+    model: NodeWithCalculator = zntrack.deps()
+    model_name: str = zntrack.params()
+
+    # ------------------------------------------------------------
+    # PDB processing functions
+    # ------------------------------------------------------------
+    @staticmethod
+    def extract_charge_and_selections(
+        pdb_path: Path,
+    ) -> tuple[float, float, float, str, str]:
+        """
+        Extract charge and selection information from PDB REMARK lines.
+
+        Parameters
+        ----------
+        pdb_path
+            Path to pdb files.
+
+        Returns
+        -------
+        tuple[int, int, int, list, list]: A tuple containing:
+
+            - total_charge: Total charge of the system.
+            - qa: Charge of selection a.
+            - qb: Charge of selection b.
+            - selection_a: Atom indices of selection a.
+            - selection_b: Atom indices of selection b.
+        """
+        total_charge = qa = qb = 0.0
+        selection_a = selection_b = ""
+
+        with open(pdb_path) as f:
+            for line in f:
+                if not line.startswith("REMARK"):
+                    if line.startswith("ATOM") or line.startswith("HETATM"):
+                        break
+                    continue
+
+                parts = line.split()
+                if len(parts) < 3:
+                    continue
+
+                tag = parts[1].lower()
+
+                if tag == "charge":
+                    total_charge = float(parts[2])
+                elif tag == "charge_a":
+                    qa = float(parts[2])
+                elif tag == "charge_b":
+                    qb = float(parts[2])
+                elif tag == "selection_a":
+                    selection_a = " ".join(parts[2:])
+                elif tag == "selection_b":
+                    selection_b = " ".join(parts[2:])
+
+        return total_charge, qa, qb, selection_a, selection_b
+
+    @staticmethod
+    def separate_protein_ligand_simple(pdb_path: Path):
+        """
+        Separate based on residue names.
+
+        Parameters
+        ----------
+        pdb_path
+            Path to pdb files.
+
+        Returns
+        -------
+        tuple[Mda.universe.atoms, Mda.universe.atoms, Mda.universe.atoms]:
+
+            - u.atoms: MDA atoms of the full system.
+            - protein_atoms: MDA atoms of the protein.
+            - ligand_atoms: MDA atoms of the ligand.
+        """
+        import MDAnalysis as Mda
```

Comment:
Is this the only time MDAnalysis is used? I think you can just read the PDB files directly with ASE, which saves a conversion later, and an extra dependency.

We can access these via `a.arrays["residuenames"]`.